packages/itwinui-react/src/core/Alert/Alert.tsx

```diff
@@ -5,104 +5,184 @@
- * <Alert type='warning'>This is a warning alert.</Alert>
- * <Alert type='negative'>This is a negative alert.</Alert>
- * <Alert
- *  type='positive'
- *  clickableText="I am a clickable text"
- *  clickableTextProps={{ href: 'https://www.example.com' }}
- * >
- *   This is a positive alert with a clickable text
- * </Alert>
- */
-export const Alert = React.forwardRef((props, ref) => {
+const AlertComponent = React.forwardRef((props, ref) => {
   const {
     children,
     className,
     type = 'informational',
-    clickableText,
-    clickableTextProps,
-    onClose,
     isSticky = false,
     ...rest
   } = props;
 
-  const StatusIcon = StatusIconMap[type];
-
   return (
     <Box
+      as='div'
       className={cx('iui-alert', className)}
       data-iui-status={type}
       data-iui-variant={isSticky ? 'sticky' : undefined}
       ref={ref}
       {...rest}
     >
-      <StatusIcon className='iui-alert-icon' />
-      <Box as='span' className='iui-alert-message'>
-        {children}
-        {clickableText && (
-          <Box
-            as='a'
-            {...clickableTextProps}
-            className={cx('iui-alert-link', clickableTextProps?.className)}
-          >
-            {clickableText}
-          </Box>
-        )}
-      </Box>
-
-      {onClose && (
-        <Box
-          as='button'
-          onClick={onClose}
-          aria-label='Close'
-          type='button'
-          className='iui-alert-button'
-        >
-          <Box as='span' className='iui-alert-button-icon' aria-hidden>
-            <SvgCloseSmall />
-          </Box>
-        </Box>
-      )}
+      <AlertContext.Provider value={{ type }}>{children}</AlertContext.Provider>
     </Box>
   );
-}) as PolymorphicForwardRefComponent<'div', AlertProps>;
+}) as PolymorphicForwardRefComponent<'div', AlertOwnProps>;
+AlertComponent.displayName = 'Alert';
+
+// ----------------------------------------------------------------------------
+// Alert.Icon component
+
+type AlertIconOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const AlertIcon = React.forwardRef((props, ref) => {
+  const { children, ...rest } = props;
+
+  const { type } = useSafeContext(AlertContext);
+
+  if (!children) {
+    const StatusIcon = StatusIconMap[type];
+    return (
+      <Icon fill={type}>
+        <StatusIcon ref={ref} {...rest} />
+      </Icon>
+    );
+  }
+
+  return (
+    <Icon ref={ref} {...rest}>
+      {children}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<'span', AlertIconOwnProps>;
+AlertIcon.displayName = 'Alert.Icon';
+
+// ----------------------------------------------------------------------------
+// Alert.Message component
+
+const AlertMessage = polymorphic.span('iui-alert-message');
+AlertMessage.displayName = 'Alert.Message';
+
+// ----------------------------------------------------------------------------
+// Alert.Action component
+
+type AlertActionOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
```

Comment:
these empty objects are no longer necessary as we are not exporting types.

```diff
- }) as PolymorphicForwardRefComponent<'a', AlertActionOwnProps>;
+ }) as PolymorphicForwardRefComponent<'a'>;
```

Reply:
Fixed